homeassistant/components/modbus/modbus.py

```diff
@@ -56,9 +62,12 @@ def modbus_setup(
         for component, conf_key in (
             (CONF_CLIMATE, CONF_CLIMATES),
             (CONF_COVER, CONF_COVERS),
+            (CONF_BINARY_SENSOR, CONF_BINARY_SENSORS),
+            (CONF_SENSOR, CONF_SENSORS),
+            (CONF_SWITCH, CONF_SWITCHES),
         ):
             if conf_key in conf_hub:
-                load_platform(hass, component, DOMAIN, conf_hub, config)
+                load_platform(hass, component, DOMAIN, conf_hub, conf_hub)
```

Comment:
The last parameter should be the whole original config.
```suggestion
                load_platform(hass, component, DOMAIN, conf_hub, config)
```